remote-method/index.js

```diff
@@ -0,0 +1,285 @@
+var chalk = require('chalk');
+
+var actions = require('../lib/actions');
+var helpers = require('../lib/helpers');
+var validateName = helpers.validateName;
+var typeChoices =helpers.getTypeChoices;
+var accepts = [];
+var returns = [];
+var https = [];
+
+module.exports = yeoman.generators.Base.extend({
+  // This generator does not track file changes via yeoman,
+  // as loopback-workspace is editing (modifying) files when
+  // saving project changes.
+  constructor: function() {
+    yeoman.generators.Base.apply(this, arguments);
+
+    this.argument('modelName', {
+      desc: 'Name of the model',
+      required: false,
+      type: String
+    });
+  },
+
+  help: function() {
+    return helpers.customHelp(this);
+  },
+
+  loadProject: actions.loadProject,
+
+  loadModels: actions.loadModels,
+
+  askForModel: function() {
+    if (this.modelName) {
+      this.modelDefinition = this.projectModels.filter(function(m) {
+        return m.name === this.modelName;
+      }.bind(this))[0];
+
+      if(!this.modelDefinition) {
+        var msg = 'Model not found: ' + this.modelName +
+                  ' Please choose from Model List:';
+        this.log(chalk.red(msg));
+      }
+    }
+
+    if (!this.modelDefinition) {
+      var done = this.async();
+      var prompts = [
+        {
+          name: 'model',
+          message: 'Select the model:',
+          type: 'list',
+          choices: this.modelNames
+        }
+      ];
+      this.prompt(prompts, function(answers) {
+        this.modelName = answers.model;
+        done();
+      }.bind(this));
+    }
+  },
+
+  findModelDefinition: function() {
+    this.modelDefinition = this.projectModels.filter(function(m) {
+        return m.name === this.modelName;
+    }.bind(this))[0];
+
+    if (!this.modelDefinition) {
+        var msg = 'Model not found: ' + this.modelName;
+        this.log(chalk.red(msg));
+        this.async()(new Error(msg));
+    }
+  },
+
+  askForParameters: function() {
+    var done = this.async();
+    var prompts = [
+      {
+        name: 'methodName',
+        message: 'Enter the remote method name:',
+        required: true,
+        validate: function(value) {
+          return validateName(value);
+        }
+      },
+      {
+        name: 'isStatic',
+        message: 'Is Static?',
+        required:true,
+        type: 'confirm',
+        default: true
+      },
+      {
+        name: 'description',
+        message: 'Description for method:'
+      }
+    ];
+    this.prompt(prompts, function(answers) {
+      this.methodName = answers.methodName;
+      this.isStatic = answers.isStatic;
+      this.description = answers.description;
+      done();
+    }.bind(this));
+  },
+
+  delimEndpoints: function() {
+    this.log('Let\'s add some Endpoints now.\n');
+  },
+
+  askForEndpoints: function() {
+    var done = this.async();
+
+    var prompts = [
+      {
+        name: 'httpPath',
+        message: 'Enter path for endpoints',
```

Comment:
Edit: on the second thought, I am not sure if "HTTP path" makes much sense.

Let's use `Enter the path of this endpoint:`